iceoryx2-bb/cxx/include/iox2/container/optional.hpp

```diff
@@ -332,23 +328,6 @@ Optional(T) -> Optional<T>;
 
 #endif
 
-#if __cplusplus >= 201703L
-// NOLINTNEXTLINE(readability-identifier-naming), for consistency with C++17 code using std::optional
-inline constexpr NulloptT nullopt { detail::NulloptTConstructorTag {} };
-#else
-namespace detail {
-template <typename = void>
-struct NulloptHelper {
-    static NulloptT nullopt;
-};
-template <>
-// NOLINTNEXTLINE(misc-definitions-in-headers), not an ODR violation because of template
-NulloptT NulloptHelper<>::nullopt = NulloptT { detail::NulloptTConstructorTag {} };
-} // namespace detail
-// NOLINTNEXTLINE(readability-identifier-naming), for consistency with C++17 code using std::optional
-static constexpr NulloptT const& nullopt = detail::NulloptHelper<>::nullopt;
-#endif
```

Comment:
@ComicSansMS I had to remove this since there were multiple definition errors in the tests and clang-tidy also complained. As workaround, I took the same approach as we had in iceoryx_hoofs and it seems to work.

Feel free to use your magic to revert my workaround :)

Reply:
Yep, you are right, this was actually broken.

The hoofs implementation does have observably different behavior, albeit in a subtle way. It boils down to [this example](https://godbolt.org/z/G6djMqfcd). This should not be a problem for the common uses of `nullopt`. I am considering to reintroduce an `#ifdef` to get the correct behavior in 17, as it should not make the situation worse, even on mixed builds. But I will ponder on that some more...